.devcontainer/Dockerfile

```diff
@@ -67,7 +67,12 @@ RUN apt-get -y install cpp-10 gcc-10 g++-10
 # clang
 RUN apt-get -y install clang-12 llvm-12
 # rust
-RUN ./install-rust.sh
+# there is a bug in vscode's handling of devcontainer contexts that means
```

Comment:
a better workaround might be to do:
```
diff --git a/.devcontainer/Dockerfile b/.devcontainer/Dockerfile
index 9a325083c..ed531f707 100644
--- a/.devcontainer/Dockerfile
+++ b/.devcontainer/Dockerfile
@@ -67,7 +67,6 @@ RUN apt-get -y install cpp-10 gcc-10 g++-10
 # clang
 RUN apt-get -y install clang-12 llvm-12
 # rust
-RUN ./install-rust.sh
 ENV PATH "/root/.cargo/bin:$PATH"

 # clang by default
diff --git a/.devcontainer/devcontainer.json b/.devcontainer/devcontainer.json
index fa6c17e38..253a6b775 100644
--- a/.devcontainer/devcontainer.json
+++ b/.devcontainer/devcontainer.json
@@ -8,6 +8,7 @@
                        "APT_MIRROR": "${localEnv:APT_MIRROR}"
                }
        },
+       "postStartCommand": "./install-rust.sh",
        // The optional 'runArgs' property can be used to specify additional runtime arguments.
        "runArgs": [
                // Uncomment the line if you will use a ptrace-based debugger like C++, Go, and Rust.
```

It's possible to use `postCreateCommand ` instead of `postStartCommand` but in the context of a dev environment, we probably just want to run that script all the time anyways

Reply:
Won't that reinstall rust in the container _every time_ you start the container? I thought the point was to cache an image that serves as a build environment.

Reply:
the scripts no-ops if there is nothing to do

Reply:
Ok but is the container _stateful_? I thought it would reinitialize itself every time it starts. I honestly don't know what sort of mess of union filesystems or temporary filesystems is involved, if that works go for it!

Reply:
yes it works - dev containers need to be explicitly reset to lose state

Reply:
I ended up opening https://github.com/stellar/stellar-core/pull/3683 as I was debugging something and there was some issues on Windows in general